pkg/http/listener.go

```diff
@@ -107,6 +96,30 @@ func (listener *httpListener) start() {
 		}
 	}
 
+	// Guess TLS protocol returns true if HTTP1.1 or
+	// HTTP2 methods are not found in the input buffer.
+	//
+	// This function if it returns true doesn't mean
+	// buffer has indeed TLS data, which needs
+	// to be further validated by using tls handshake.
+	guessTLSProtocol := func(buf []byte) bool {
+		// Check for HTTP2 methods first.
+		for _, m := range defaultHTTP2Methods {
+			if strings.HasPrefix(string(buf), m) {
+				return false
+			}
+		}
+
+		// Check for HTTP1 methods.
+		for _, m := range defaultHTTP1Methods {
+			if strings.HasPrefix(string(buf), m) {
+				return false
+			}
+		}
+
+		return true
```

Comment:
There is no point in handling http1/http2 methods separately in `httpListener` case